stripe2qbo/api/routers/transaction_router.py

```diff
@@ -0,0 +1,36 @@
+from typing import Annotated
+
+from sqlalchemy.orm import Session
+from fastapi import APIRouter, Depends
+
+from stripe2qbo.api.dependencies import get_current_user, get_db
+from stripe2qbo.db.models import User
+
+from stripe2qbo.db.schemas import TransactionSync
+
+router = APIRouter(
+    prefix="/transaction",
+    tags=["transaction"],
+)
+
+
+@router.get("/")
+async def get_all_transactions(
+    user: Annotated[User, Depends(get_current_user)],
+    db: Annotated[Session, Depends(get_db)],
+) -> TransactionSync:
+    return db.query(TransactionSync).filter(TransactionSync.user_id == user.id).all()
```

Comment:
This is not going to work. You need a SQLAlchemy model for querying the DB like this

I'm surprised pre-commit didn't catch this... if you run `mpy .` you'll get a bunch of type errors. I'll look into the bug with pre-commit.

Anyway, there are two models you need. It's kind of annoying. But necessary (for now). Just import them as different names.

1. A SQLAlchemy model to act as an ORM. This you will use for communicating with the database
2. A Pydantic model to play nice with FastAPI and the rest of the app

So you first fetch the data as a TransactionSyncORM, then return it as a TransactionSync (Pydantic model)

Also, remember you want to return a list, so update the return type accordingly
e.g.
```python
from stripe2qbo.db.models import TransactionSync as TransactionSyncORM` 
from stripe2qbo.db.schemas import TransactionSync

@router.get("/")
async def get_all_transactions(
    user: Annotated[User, Depends(get_current_user)],
    db: Annotated[Session, Depends(get_db)],
) -> List[TransactionSync]:
    transactions = db.query(TransactionSyncORM).filter(TransactionSyncORM.user_id == user.id).all()
    return [TransactionSync.model_validate(tsx, { "from_attributes": True }) for tsx in transactions]
```